uint32 headerSize
uint32 dataSize
uint8 noBuffers